src/check.py

```diff
@@ -89,14 +138,47 @@ def _update_action_problem(action: UpdateAction) -> Problem | None:
     return problem
 
 
-def conflicts(actions: Iterable[AnyAction]) -> Iterator[Problem]:
+def conflicts(
+    actions: Iterable[AnyAction], repository: Client, user_inputs: UserInputs
+) -> Iterator[Problem]:
     """Check whether actions have any content conflicts.
 
     Args:
         actions: The actions to check.
+        repository: Client for repository interactions.
+        user_inputs: Configuration from the user.
 
     Yields:
         A problem for each action with a conflict
     """
+    track_paths_with_diff = TrackPathsWithDiff()
+    commit_tagged = repository.is_same_commit(
+        tag=constants.DISCOURSE_AHEAD_TAG, commit=user_inputs.commit_sha
+    )
+
     update_actions = filter(_is_update_action, actions)
-    yield from filter(None, (_update_action_problem(action) for action in update_actions))
+    tracked_update_actions = side_effect(track_paths_with_diff.process, update_actions)
+
+    yield from filter(None, (_update_action_problem(action) for action in tracked_update_actions))
```

Comment:
can we put a check that if some "actual" conflicts have been found, we just return these problems only? I mean, I don't think there is much of a point of reporting also soft conflicts, given they are a subset of hard conflict (if there is a hard conflict it is obvious that there is also a soft one)

Reply:
Good point, changed so that these are only yielded if there are no page by page conflicts